packages/internal/src/priority-senders.js

```diff
@@ -0,0 +1,104 @@
+import { E, Far } from '@endo/far';
+
+const { Fail, quote: q } = assert;
+
+const PRIORITY_SENDERS_NAMESPACE_RE = /^[a-zA-Z0-9_-]{1,50}$/;
+
+const assertNormalizedNamespace = namespace => {
+  if (!namespace.match(PRIORITY_SENDERS_NAMESPACE_RE)) {
+    throw Fail`invalid namespace ${q(namespace)}`;
+  }
+};
+
+/** @type {(namespace: string) => string} */
+export const normalizeSenderNamespace = namespace => {
+  const candidate = namespace.replace(/[ ,()]/g, '_');
+  assertNormalizedNamespace(candidate);
+  return candidate;
+};
```

Comment:
This can result in error messages complaining about already-transformed input.
```suggestion
/** @type {(namespace: string) => string} */
export const normalizeSenderNamespace = namespace => {
  const candidate = namespace.replace(/[ ,()]/g, '_');
  PRIORITY_SENDERS_NAMESPACE_RE.test(candidate) ||
    Fail`invalid namespace ${q(namespace)}`;
  return candidate;
};
```

Reply:
Good point, thanks